modyn/supervisor/internal/grpc_handler.py

```diff
@@ -200,18 +202,52 @@ def register_pipeline_at_selector(self, pipeline_config: dict) -> int:
         if not self.connected_to_selector:
             raise ConnectionError("Tried to register pipeline at selector, but no connection was made.")
 
+        if "config" in pipeline_config["model"]:
+            model_config = json.dumps(pipeline_config["model"]["config"])
+        else:
+            model_config = "{}"
+
+        model_storage_config = pipeline_config["model_storage"]
+        incremental_model_strategy: Optional[StrategyConfig] = None
+        full_model_interval: Optional[int] = None
+        if "incremental_model_strategy" in model_storage_config:
+            incremental_strategy = model_storage_config["incremental_model_strategy"]
+            incremental_model_strategy = self.get_model_strategy(incremental_strategy)
+            full_model_interval = (
+                incremental_strategy["full_model_interval"] if "full_model_interval" in incremental_strategy else None
+            )
+
         pipeline_id = self.selector.register_pipeline(
             RegisterPipelineRequest(
                 num_workers=pipeline_config["training"]["dataloader_workers"],
                 selection_strategy=SelectorJsonString(
                     value=json.dumps(pipeline_config["training"]["selection_strategy"])
                 ),
+                model_class_name=pipeline_config["model"]["id"],
+                model_configuration=SelectorJsonString(value=model_config),
+                amp=pipeline_config["training"]["amp"] if "amp" in pipeline_config["training"] else False,
+                model_storage_policy=ModelStoragePolicyInfo(
+                    full_model_strategy_config=self.get_model_strategy(model_storage_config["full_model_strategy"]),
+                    incremental_model_strategy_config=incremental_model_strategy,
+                    full_model_interval=full_model_interval,
+                ),
             )
         ).pipeline_id
 
         logger.info(f"Registered pipeline {pipeline_config['pipeline']['name']} at selector with ID {pipeline_id}")
         return pipeline_id
 
+    @staticmethod
+    def get_model_strategy(strategy_config: dict) -> StrategyConfig:
```

Comment:
maybe also here implement a constructor on StrategyConfig

Reply:
Similar to above. `StrategyConfig` is a message of the selector's grpc interface definition.